### PR TITLE
#2 define car prototype

### PR DESCRIPTION
**Pull request by ZsoltToth**
Please be so kind to review my code.

### CODE REVIEW

File: cars/core/src/main/java/hu/uni/miskolc/iit/webdevelopment/examples/cars/model/Car.java [hunk 1]
```diff
@@ -47,13 +98,5 @@ public boolean equals(Object obj) {
...
-    @Override
-    public String toString() {
-        return "Car{" +
-                "plateNo='" + plateNo + '\'' +
-                ", brand='" + brand + '\'' +
-                ", carModel='" + carModel + '\'' +
-                ", color=" + color +
-                '}';
-    }
+
```
Comment (judit0310) [inline cars/core/src/main/java/hu/uni/miskolc/iit/webdevelopment/examples/cars/model/Car.java]: Missing toString

File: cars/core/src/main/java/hu/uni/miskolc/iit/webdevelopment/examples/cars/model/CarPrototype.java [hunk 2]
```diff
@@ -0,0 +1,113 @@
+package hu.uni.miskolc.iit.webdevelopment.examples.cars.model;
+
+import java.util.Calendar;
+import java.util.GregorianCalendar;
```
Comment (judit0310) [inline cars/core/src/main/java/hu/uni/miskolc/iit/webdevelopment/examples/cars/model/CarPrototype.java]: Why import these?

File: cars/core/src/main/java/hu/uni/miskolc/iit/webdevelopment/examples/cars/model/CarProducer.java [hunk 3]
```diff
@@ -0,0 +1,10 @@
+package hu.uni.miskolc.iit.webdevelopment.examples.cars.model;
+
+/**
+ * Created by tothzs on 2017.11.24..
+ *
+ * List of car manufacturers.
+ */
+public enum CarProducer {
+    SUZUKI,OPEL,AUDI,FERRARI, TESLA, CITROEN, FIAT
```
Comment (judit0310) [inline cars/core/src/main/java/hu/uni/miskolc/iit/webdevelopment/examples/cars/model/CarProducer.java]: Missing producers

File: cars/core/src/main/java/hu/uni/miskolc/iit/webdevelopment/examples/cars/model/CarPrototype.java [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+     */
+    private int doorNumber;
+    /**
+     * Performance of the engine in kW.
+     */
+    private double enginePerformance;
+
+
+
+    public CarPrototype(CarProducer producer, String model, CarBodyStyle bodyStyle, int yearOfIntroduction, String note, int crubWeight, int grossVehicleWeight, int doorNumber, double enginePerformance) {
```
Comment (judit0310) [inline cars/core/src/main/java/hu/uni/miskolc/iit/webdevelopment/examples/cars/model/CarPrototype.java]: negative door number for example?

File: cars/core/src/main/java/hu/uni/miskolc/iit/webdevelopment/examples/cars/service/CarPrototypeManagerService.java [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+
+    /**
+     *
+     * @param bodyStyle selected body style
+     * @return collection of the prototypes with the selected body style.
+     */
+    Collection<CarPrototype> listCarPrototypes(CarBodyStyle bodyStyle);
+
+    /**
+     * Recrod a new Car Prototype into the system.
```
Comment (judit0310) [inline cars/core/src/main/java/hu/uni/miskolc/iit/webdevelopment/examples/cars/service/CarPrototypeManagerService.java]: *Record

**Review by judit0310 — CHANGES_REQUESTED** [review]
Please be so kind to ...

File: cars/core/src/main/java/hu/uni/miskolc/iit/webdevelopment/examples/cars/model/Car.java (hunk 1 above)
Comment (ZsoltToth) [reply cars/core/src/main/java/hu/uni/miskolc/iit/webdevelopment/examples/cars/model/Car.java]: done

File: cars/core/src/main/java/hu/uni/miskolc/iit/webdevelopment/examples/cars/model/CarPrototype.java (hunk 2 above)
Comment (ZsoltToth) [reply cars/core/src/main/java/hu/uni/miskolc/iit/webdevelopment/examples/cars/model/CarPrototype.java]: unused import have been removed

File: cars/core/src/main/java/hu/uni/miskolc/iit/webdevelopment/examples/cars/model/CarProducer.java (hunk 3 above)
Comment (ZsoltToth) [reply cars/core/src/main/java/hu/uni/miskolc/iit/webdevelopment/examples/cars/model/CarProducer.java]: as you wish

File: cars/core/src/main/java/hu/uni/miskolc/iit/webdevelopment/examples/cars/model/CarPrototype.java (hunk 4 above)
Comment (ZsoltToth) [reply cars/core/src/main/java/hu/uni/miskolc/iit/webdevelopment/examples/cars/model/CarPrototype.java]: data type of door number has been changed to enum

File: cars/core/src/main/java/hu/uni/miskolc/iit/webdevelopment/examples/cars/model/CarProducer.java [hunk 6]
```diff
@@ -2,9 +2,684 @@
 
 /**
  * Created by tothzs on 2017.11.24..
- *
+ * <p>
  * List of car manufacturers.
  */
 public enum CarProducer {
```
Comment (judit0310) [inline cars/core/src/main/java/hu/uni/miskolc/iit/webdevelopment/examples/cars/model/CarProducer.java]: :1st_place_medal:

**Review by judit0310 — APPROVED** [review]

**Pull request merged**